### PR TITLE
Patrick v chung master

### PR DESCRIPTION
**Pull request by wceichelb**

### CODE REVIEW

File: app/helpers/application_helper.rb [hunk 1]
```diff
@@ -6,7 +6,7 @@ def nav_link_to text, link, opts={}
     end
 
     def nav_link_active? link
-        request.path == link
+        request.fullpath == link
```
Comment (wceichelb) [inline app/helpers/application_helper.rb]: currently, with extra menu grouping items defined in the settings, if you click on one link, `class="active"` is applied to each of the `li` that point to the `ls_reports` path, even though `request.fullpath == /ls_reports?grouping=whatever`.

@hattwj any idea why the change to `fullpath` doesn't fix the problem?

File: config/settings.yml.example [hunk 2]
```diff
@@ -8,7 +8,15 @@ defaults: &defaults
         name: Some Site
         name_html: Test
         logo: none
-        titles: 
+        menu_groups:
```
Comment (hattwj) [inline config/settings.yml.example]: Am I reading this correctly? It looks like we are just adding empty keys.

Comment (wceichelb) [inline config/settings.yml.example]: in the example file, yes. 

the idea is to give an admin the option to define extra grouping options in the _nav_header by defining them here. i didn't want to hard code it in the app itself, though, so if the keys are left blank, we fall through to the previous behavior.

the impetus behind this is that some of members of the higher permission_groups end up having access to a big long list of surveys. evidently there've been requests for a way to get right to more relevant surveys.

Comment (hattwj) [inline config/settings.yml.example]: Ahh, i see. So the new surveys are grouped via this option instead. I'm a little confused as we already have the ability to group surveys. How was the old way broken?

Comment (wceichelb) [inline config/settings.yml.example]: image


here's an example. i only have one member of the med18 grouping in my test db, but you get the idea. if i had more in the db they'd show up in the list. the grouping is just passed in the url.

side note, the other line i pinged you on in this PR is illustrated here as well. all three links to /ls_reports* are active for some reason.

File: app/helpers/application_helper.rb (hunk 1 above)
Comment (hattwj) [inline app/helpers/application_helper.rb]: Best i can tell is because it doesn't include query strings in the match.

File: config/settings.yml.example (hunk 2 above)
Comment (hattwj) [inline config/settings.yml.example]: If we were to add a 'pin_to_nav' on the permission group it might be a good way to bypass this. I am a little concerned over the effect this might have on performance.

File: app/models/user.rb [hunk 3]
```diff
@@ -270,4 +270,17 @@ def institution
     def to_param
         username.parameterize
     end
+
+    def menu_has_groupings?
+      permissions = Settings.site['menu_groups_permissions'] || nil
+      if Settings.site['menu_groups']
+        if permissions
+          self.admin_or_higher? or permissions.include?(self.permission_group.title)
```
Comment (hattwj) [inline app/models/user.rb]: This might belong in the Abilities model. Think something like can?(:read, MenuGroup)
